src/Internals/Map.elm

```diff
@@ -144,6 +160,15 @@ withPolygons polygons (Map map) =
 
 
 
+-- Plugins
+
+
+withDrawingTool : DrawingTool.State -> DrawingTool.Config msg -> Map msg -> Map msg
+withDrawingTool state config (Map ({ plugins } as map)) =
```

Comment:
I'm not a big fan of this `config` field.
In my experience in paack-ui every time we ended creating it we got a point where API is likely to break.

My suggestion is to break this in: `withDrawingToolState` and `withDrawingStateCompletedFeedback`

Reply:
humm, but then if we add more features related to the drawing tool this will bring more with* to the map module right? 🤔 

What do you think about having a  builder constructor for the events as well like, 

`DrawingTool.events |> DrawingTool.withOnPolygonCompletedEvent msg`

In reality another question here is if we want to have this event as a mandatory event or not, because if it is the case , it would be something like `DrawingTool.events onPolygonCompletedMsg`

Reply:
:thinking:  yeah, I prefer the "events" instead of "config".

>  if we want to have this event as a mandatory event or not

Is there any other interesting listener besides "polygon completed"? Because if there is, I would say it should not be mandatory.